Sources/TuistGenerator/Mappers/ExplicitDependencyTargetMapper.swift

```diff
@@ -0,0 +1,60 @@
+import Foundation
+import TuistCore
+import TuistGraph
+import TuistSupport
+
+/// A target mapper that updates
+public struct ExplicitDependencyTargetMapper: TargetMapping {
+    public init() {}
+    
+    public func map(target: Target) throws -> (Target, [SideEffectDescriptor]) {
+        let movedProductNames = target.dependencies.compactMap {
+            switch $0 {
+            case
+                let .target(name: name, condition: _),
+                let .project(target: name, path: _, condition: _):
+                return name
+            case .library, .framework, .package, .sdk, .xcframework, .xctest:
+                return nil
+            }
+        }
+            
+        let frameworkSearchPaths = movedProductNames.map {
+            "$(CONFIGURATION_BUILD_DIR)$(TARGET_BUILD_SUBPATH)/\($0)"
+        }
```

Comment:
What about turning this one into a graph mapper and having something along the lines of:

```swift
let directDependencyTargets = GraphTraverser(graph: graph).directTargetDependencies(name: target.name, project: project)
let frameworkSearchPaths = directDependencyTargets.map(\.target.buildProductDirectoryBuildSetting)
```